src/interpreter/initialization.rs

```diff
@@ -40,7 +37,10 @@ where
             .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
 
         // Set initial unused balances
-        let free_balances = Self::initial_free_balances(&tx)?;
+        let free_balances = checked_tx.balances;
+        // Clear balance indexes in case vm is being re-initialized
+        self.unused_balance_index.clear();
```

Comment:
Hmmm. Is this necessary? Predicates can't actually modify the free balances since they can't send coins anywhere. Only scripts will, and scripts are only executed once per tx.